core/ui/TestCases/DefaultTemplate.tid

```diff
@@ -2,6 +2,9 @@ title: $:/core/ui/testcases/DefaultTemplate
 code-body: yes
 
 \whitespace trim
+
+\function tf.state() [<qualify "$:/state/testcase">]
+
 \procedure linkcatcherActions()
 <%if [<navigateTo>has[title]] %>
 	<$qualify title=<<state>> name="qualifiedState">
```

Comment:
Link clicks are not working for me (see `TestCases/TranscludeWidget/SimpleTransclusion` for an example with links).

Maybe this will fix it, but I'm not sure. I don't understand why the original code has both a qualify macro call and a qualify widget call:

```suggestion
	<$qualify title=<<tf.state>> name="qualifiedState">
```

Reply:
The linkcatcher code is needed to switch tabs if links are clicked in the Output. qualify-widget is needed to define a "name" parameter

Reply:
I understand linkcatcher. Here is my confusion:

`\function tf.state() [<qualify "$:/state/testcase">]` <== qualify macro

`<$qualify title=<<tf.state>> name="qualifiedState">` <== qualify widget

Why are both needed? Maybe that is already what you are explaining and I missed the point.

And now that I think about it, the testcases are executed in a "sandbox" tiddler store. I'm not sure why qualify is needed. Just use a static, hard-coded tiddler name for the tab state? But this PR is only about refactoring, not making changes. I'm just raising the question.

Reply:
From  line 112 to 121:

```
\procedure testcase-source()
<div class="tc-test-case-source">
	<$macrocall $name="tabs"
		tabsList=<<testcase-tabsList>>
		state=<<tf.state>>
		default="Output"
		template="$:/core/ui/testcases/DefaultTemplate/SourceTabs"
	/>
</div>
\end
```

The `tabs` macro was used, which appends the qualify number to `tf.state` (which was qualified) as the state tiddler title. In order to get the state tiddler, we have to qualify `$:/state/testcase` twice, which requires using the qualify widget.

Reply:
>Why are both needed? Maybe that is already what you are explaining and I missed the point.

You are right. IMO - That's a bug. If the tabs-macro gets a `state` parameter it internally uses the qualify macro. The `explicitState` parameter should be used. So the linkcatcher code can be much simpler. 

**and**

>And now that I think about it, the testcases are executed in a "sandbox" tiddler store. I'm not sure why qualify is needed.

I think if there are example tiddlers that contain a `tabs` macro we may have name clashes. But we would need to test that in detail.

>But this PR is only about refactoring, not making changes. I'm just raising the question.

That's right. **We can improve the code after this PR has been merged.**